src/controller.rs

```diff
@@ -166,33 +285,20 @@ where
 
     let rc = ReconciliationContext::new(context.client.clone(), resource.clone());
 
-    let state = strategy.init_reconcile_state(rc);
-    let futures = state.reconcile_operations();
-
-    for future in futures {
-        let result = future.await;
-
-        match result {
-            Ok(ReconcileFunctionAction::Continue) => {
-                trace!("Reconciler loop: Continue");
-            }
-            Ok(ReconcileFunctionAction::Done) => {
-                trace!("Reconciler loop: Done");
-                break;
-            }
-            Ok(ReconcileFunctionAction::Requeue(duration)) => {
-                trace!(?duration, "Reconciler loop: Requeue");
-                return Ok(ReconcilerAction {
-                    requeue_after: Some(duration),
-                });
-            }
-            Err(err) => {
-                error!(?err, "Error reconciling");
-                return Ok(reconcile::create_requeuing_reconciler_action(
-                    Duration::from_secs(30),
-                ));
-                // TODO: Make configurable
-            }
+    let mut state = strategy.init_reconcile_state(rc);
+    let result = state.reconcile().await;
+    match result {
+        Ok(ReconcileFunctionAction::Requeue(duration)) => {
+            trace!(?duration, "Reconciler loop: Requeue");
+            return Ok(ReconcilerAction {
+                requeue_after: Some(duration),
+            });
+        }
+        Ok(action) => {
+            trace!("Reconciler loop: {:?}", action);
+        }
+        Err(err) => {
+            error!("Error reconciling [{:?}]", err);
```

Comment:
Previously we requeued for an error, was this a conscious decision to do it differently?

Reply:
I'd like to say yes.


But that'd be a lie.

Reply:
Fixed